rewriter/initializer.cc

```diff
@@ -0,0 +1,128 @@
+#include "rewriter/initializer.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/core.h"
+
+#include <utility>
+
+using namespace std;
+namespace sorbet::rewriter {
+
+// this only looks for sigs of the 'correct' shape (i.e. sigs with blocks that contain a send that also end with `void`)
+bool isSig(const ast::Send *send) {
+    if (send->fun != core::Names::sig()) {
+        return false;
+    }
+    if (send->block.get() == nullptr) {
+        return false;
+    }
+    auto nargs = send->args.size();
+    if (nargs != 0 && nargs != 1) {
+        return false;
+    }
+    auto block = ast::cast_tree<ast::Block>(send->block.get());
+    ENFORCE(block);
+    auto body = ast::cast_tree<ast::Send>(block->body.get());
+    if (!body) {
+        return false;
+    }
+    if (body->fun != core::Names::void_()) {
+        return false;
+    }
+
+    return true;
+}
+
+bool isCopyableType(const ast::Expression *typeExpr) {
+    auto send = ast::cast_tree_const<ast::Send>(typeExpr);
+    if (send && send->fun == core::Names::typeParameter()) {
```

Comment:
Can you link to #1715, because this is something we could remove once that bug is fixed.

Reply:
Done!